src/lib.rs

```diff
@@ -150,6 +166,61 @@ pub mod error;
 mod session;
 use crate::session::{Cmd, Session};
 
+/// Builder pattern for Client
+#[derive(Default, Clone, Debug)]
+pub struct ClientBuilder<C>
+where
+    C: connect::Connect + Send + Sync + Clone + Unpin,
+{
+    capabilities: Option<webdriver::capabilities::Capabilities>,
+    connector: C,
+}
+
+#[cfg(feature = "rustls-tls")]
+impl ClientBuilder<hyper_rustls::HttpsConnector<hyper::client::HttpConnector>> {
+    /// Create a builder with Rustls connector, available with feature `rustls-tls`
+    pub fn rustls() -> Self {
+        Self::new(hyper_rustls::HttpsConnector::with_native_roots())
+    }
+}
+
+#[cfg(feature = "openssl-tls")]
+impl ClientBuilder<hyper_tls::HttpsConnector<hyper::client::HttpConnector>> {
+    /// Create a builder with OpenSSL connector, available with feature `openssl-tls`
+    pub fn openssl() -> Self {
+        Self::new(hyper_tls::HttpsConnector::new())
+    }
+}
+impl<C> ClientBuilder<C>
+where
+    C: connect::Connect + Send + Sync + Clone + Unpin + 'static,
+{
+    /// Create a new builder with TLS connector
+    /// # Arguments
```

Comment:
In Rust, arguments aren't usually documented as lists (unlike in Java for example). Instead, they are more commonly incorporated into explanatory text. I'd suggest:
```rust
/// Create a builder using `connector` as the TLS connector.
```